In [226]:
import pandas as pd
from sqlalchemy import create_engine
import psycopg2
import numpy as np

In [2]:
# establish a database connection
engine = create_engine("postgres+psycopg2://postgres:postgres@localhost:5432/lahman_baseball")
# Replace `<lahman_baseball_database_name>` with the actual name of your lahman baseball database as it appears in pgadmin

# use the connection to run a query using pandas!
batting = pd.read_sql("SELECT * FROM batting;", con=engine)

### a. Find all players in the database who played at Vanderbilt University.
### b. Create a list showing each player’s first and last names as well as the total salary they earned in the major leagues.
### c. Sort this list in descending order by the total salary earned.
### d. Which Vanderbilt player earned the most money in the majors?

In [3]:
schools = pd.read_sql("SELECT * FROM schools;", con=engine)

In [4]:
schools[schools['schoolcity'] == 'Nashville']

schoolid                  schoolname schoolcity schoolstate schoolnick
74     belmont          Belmont University  Nashville          TN        USA
547   lipscomb         Lipscomb University  Nashville          TN        USA
992     tennst  Tennessee State University  Nashville          TN        USA
1110     vandy       Vanderbilt University  Nashville          TN        USA

In [5]:
collegeplaying = pd.read_sql('SELECT * FROM collegeplaying;', con=engine)

In [6]:
vandy_players = collegeplaying[collegeplaying['schoolid'] == 'vandy']

In [7]:
vandy_players.shape

(65, 3)

In [8]:
salaries = pd.read_sql('SELECT * FROM salaries;', con=engine)

In [9]:
people = pd.read_sql('SELECT * FROM people;', con=engine)

In [10]:
vandy_names = pd.merge(vandy_players, people, on = 'playerid')
vandy_names = vandy_names.drop_duplicates('playerid')

In [11]:
vandy_names = vandy_names[['playerid', 'namefirst', 'namelast']]

In [12]:
vandy_names.shape

(24, 3)

In [13]:
vandy_salaries = pd.merge(vandy_names, salaries, on = 'playerid')
vandy_salaries.playerid.unique()

array(['alvarpe01', 'baxtemi01', 'chrisni01', 'corajo01', 'flahery01',
       'grayso01', 'katama01', 'lewisje01', 'madissc01', 'minormi01',
       'pauljo01', 'priceda01', 'priorma01', 'sandesc01', 'sowerje01'],
      dtype=object)

In [14]:
vandy_salaries['name'] = vandy_salaries['namefirst'] + ' ' + vandy_salaries['namelast']
vandy_salaries.shape

(74, 8)

In [15]:
vandy_salaries = vandy_salaries.groupby('name').sum()

In [16]:
vandy_salaries = vandy_salaries[['salary']]

In [17]:
vandy_salaries.sort_values(by = 'salary', ascending=False)

salary
name                       
David Price      81851296.0
Pedro Alvarez    20681704.0
Mark Prior       12800000.0
Scott Sanderson  10750000.0
Mike Minor        6837500.0
Joey Cora         5622500.0
Ryan Flaherty     4061000.0
Josh Paul         2640000.0
Mike Baxter       2094418.0
Sonny Gray        1542500.0
Jensen Lewis      1234000.0
Matt Kata         1060000.0
Nick Christiani    500000.0
Jeremy Sowers      384800.0
Scotti Madison     135000.0

### Using the fielding table, group players into three groups based on their position: label players with position OF as "Outfield", those with position "SS", "1B", "2B", and "3B" as "Infield", and those with position "P" or "C" as "Battery". Determine the number of putouts made by each of these three groups in 2016.

In [18]:
fielding = pd.read_sql('SELECT * FROM fielding;', con=engine)
fielding['po'].sum()

11408383

In [19]:
fielding = fielding[fielding['yearid'] == 2016]
fielding

playerid  yearid  stint teamid lgid pos    g     gs  innouts    po  \
134862   abadfe01    2016      1    MIN   AL   P   39    0.0    102.0     0   
134863   abadfe01    2016      2    BOS   AL   P   18    0.0     38.0     0   
134864  abreujo02    2016      1    CHA   AL  1B  152  152.0   4067.0  1243   
134865  achteaj01    2016      1    LAA   AL   P   27    0.0    113.0     2   
134866  ackledu01    2016      1    NYA   AL  1B   13   10.0    255.0    80   
...           ...     ...    ...    ...  ...  ..  ...    ...      ...   ...   
136810  zobribe01    2016      1    CHN   NL  2B  119  113.0   2929.0   177   
136811  zobribe01    2016      1    CHN   NL  OF   46   29.0    859.0    43   
136812  zobribe01    2016      1    CHN   NL  SS    1    0.0      6.0     0   
136813  zuninmi01    2016      1    SEA   AL   C   52   48.0   1331.0   400   
136814   zychto01    2016      1    SEA   AL   P   12    0.0     41.0     0   

            a     e     dp   pb  wp    sb   cs  zr  
134862    3.0   0.0    1.0  NaN NaN   NaN  NaN NaN  
134863    1.0   0.0    0.0  NaN NaN   NaN  NaN NaN  
134864   84.0  10.0  131.0  NaN NaN   NaN  NaN NaN  
134865    4.0   0.0    0.0  NaN NaN   NaN  NaN NaN  
134866    7.0   0.0    7.0  NaN NaN   NaN  NaN NaN  
...       ...   ...    ...  ...  ..   ...  ...  ..  
136810  250.0   7.0   52.0  NaN NaN   NaN  NaN NaN  
136811    1.0   0.0    0.0  NaN NaN   NaN  NaN NaN  
136812    0.0   0.0    0.0  NaN NaN   NaN  NaN NaN  
136813   15.0   0.0    0.0  3.0 NaN  19.0  7.0 NaN  
136814    0.0   1.0    0.0  NaN NaN   NaN  NaN NaN  

[1953 rows x 18 columns]

In [20]:
fielding.loc[fielding['pos'] == 'OF', 'Label'] = 'Outfield'
fielding.loc[fielding['pos'].isin(["SS", "1B", "2B", "3B"]), 'Label'] = 'Infield'
fielding.loc[fielding['pos'].isin(["P", "C"]), 'Label'] = 'Battery'

In [21]:
fielding = fielding[['po', 'Label']]

In [22]:
fielding = fielding.groupby('Label').sum()

In [23]:
fielding

po
Label          
Battery   41424
Infield   58934
Outfield  29560

"Battery"	41424
"Infield"	58934
"Outfield"	29560

### 3. a. From 1970 – 2016, what is the largest number of wins for a team that did not win the world series?

In [25]:
teams = pd.read_sql('SELECT * FROM teams;', con=engine)

In [26]:
teams = teams[teams['yearid'] >= 1970]

In [27]:
teams_wsno = teams[teams['wswin'] == 'N']

In [28]:
teams_wsno = teams_wsno[['teamid', 'w', 'wswin']]

In [29]:
teams_wsno.sort_values(by = 'w', ascending = False)

teamid    w wswin
2379    SEA  116     N
2267    ATL  106     N
2471    SLN  105     N
2125    ATL  104     N
2012    OAK  104     N
...     ...  ...   ...
1825    MIN   41     N
1829    NYN   41     N
1833    SDN   41     N
1818    CHN   38     N
1838    TOR   37     N

[1220 rows x 3 columns]

### b. What is the smallest number of wins for a team that did win the world series?

In [30]:
teams_wsyes = teams[teams['wswin'] == 'Y']

In [31]:
teams_wsyes = teams_wsyes[['teamid', 'yearid', 'w', 'wswin']]

In [32]:
teams_wsyes.sort_values(by = 'w').head()

teamid  yearid   w wswin
1824    LAN    1981  63     Y
2530    SLN    2006  83     Y
1981    MIN    1987  85     Y
2343    NYA    2000  87     Y
2769    SFN    2014  88     Y

### c. Doing this will probably result in an unusually small number of wins for a world series champion – determine why this is the case. (Player strike in 1981)
### d. Then redo your query, excluding the problem year.

In [33]:
teams_wsyes = teams_wsyes[teams_wsyes['yearid'] != 1981]

In [34]:
teams_wsyes.sort_values(by = 'w').head()

teamid  yearid   w wswin
2530    SLN    2006  83     Y
1981    MIN    1987  85     Y
2343    NYA    2000  87     Y
2769    SFN    2014  88     Y
1654    OAK    1974  90     Y

### e. How often from 1970 – 2016 was it the case that a team with the most wins also won the world series?
### f. What percentage of the time?

In [66]:
maxw = teams.groupby('yearid').w.max().to_frame().reset_index()
maxw.head()

yearid    w
0    1970  108
1    1971  101
2    1972   96
3    1973   99
4    1974  102

In [67]:
merged = teams_wsyes.merge(maxw, on = 'yearid')
merged.head()

teamid  yearid  w_x wswin  w_y
0    BAL    1970  108     Y  108
1    PIT    1971   97     Y  101
2    OAK    1972   93     Y   96
3    OAK    1973   94     Y   99
4    OAK    1974   90     Y  102

In [69]:
merged.columns = ['team', 'year', 'team_wins', 'wswin', 'year_max_wins']
merged.head()

team  year  team_wins wswin  year_max_wins
0  BAL  1970        108     Y            108
1  PIT  1971         97     Y            101
2  OAK  1972         93     Y             96
3  OAK  1973         94     Y             99
4  OAK  1974         90     Y            102

In [70]:
w_list = []
for row_ind, row_values in merged.iterrows():
    if row_values['team_wins'] == row_values['year_max_wins']:
        w_list.append(1)
    else: 
        w_list.append(0)

In [71]:
w_list

[1,
 0,
 0,
 0,
 0,
 1,
 1,
 0,
 1,
 0,
 0,
 0,
 0,
 1,
 0,
 1,
 0,
 0,
 1,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 1,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 0,
 1,
 0,
 1,
 0,
 0,
 0,
 1,
 0,
 0,
 1]

In [59]:
merged['w_list'] = w_list

In [65]:
(merged.w_list.sum()/merged.w_list.count())*100

26.666666666666668

### 4. Which managers have won the TSN Manager of the Year award in both the National League (NL) and the American League (AL)? Give their full name and the teams that they were managing when they won the award.

In [74]:
awardsmanagers = pd.read_sql('SELECT * FROM awardsmanagers;', con=engine)

In [76]:
awardsmanagers.head()

playerid                    awardid  yearid lgid   tie notes
0  larusto01  BBWAA Manager of the Year    1983   AL  None  None
1  lasorto01  BBWAA Manager of the Year    1983   NL  None  None
2  andersp01  BBWAA Manager of the Year    1984   AL  None  None
3   freyji99  BBWAA Manager of the Year    1984   NL  None  None
4    coxbo01  BBWAA Manager of the Year    1985   AL  None  None

In [103]:
managers = pd.read_sql('SELECT * FROM managers;', con=engine)

In [89]:
AL = awardsmanagers[(awardsmanagers['awardid'] == 'TSN Manager of the Year') & (awardsmanagers['lgid'] == 'AL')]
NL = awardsmanagers[(awardsmanagers['awardid'] == 'TSN Manager of the Year') & (awardsmanagers['lgid'] == 'NL')]

In [144]:
both = AL.merge(NL, on= 'playerid')

In [145]:
both

playerid                awardid_x  yearid_x lgid_x tie_x notes_x  \
0  johnsda02  TSN Manager of the Year      1997     AL  None    None   
1  leylaji99  TSN Manager of the Year      2006     AL  None    None   
2  leylaji99  TSN Manager of the Year      2006     AL  None    None   
3  leylaji99  TSN Manager of the Year      2006     AL  None    None   

                 awardid_y  yearid_y lgid_y tie_y notes_y  
0  TSN Manager of the Year      2012     NL  None    None  
1  TSN Manager of the Year      1988     NL  None    None  
2  TSN Manager of the Year      1990     NL  None    None  
3  TSN Manager of the Year      1992     NL  None    None

In [112]:
manager_awards = awardsmanagers[(awardsmanagers['playerid'].isin(tuple(both['playerid'])) & (awardsmanagers['awardid'] == 'TSN Manager of the Year'))]
manager_awards = manager_awards.drop(columns = ['tie', 'notes'])
manager_awards

playerid                  awardid  yearid lgid
118  leylaji99  TSN Manager of the Year    1988   NL
121  leylaji99  TSN Manager of the Year    1990   NL
126  leylaji99  TSN Manager of the Year    1992   NL
136  johnsda02  TSN Manager of the Year    1997   AL
154  leylaji99  TSN Manager of the Year    2006   AL
166  johnsda02  TSN Manager of the Year    2012   NL

In [113]:
both_names = pd.merge(manager_awards, people[['playerid','namefirst', 'namelast']], on='playerid')

In [114]:
both_names

playerid                  awardid  yearid lgid namefirst namelast
0  leylaji99  TSN Manager of the Year    1988   NL       Jim  Leyland
1  leylaji99  TSN Manager of the Year    1990   NL       Jim  Leyland
2  leylaji99  TSN Manager of the Year    1992   NL       Jim  Leyland
3  leylaji99  TSN Manager of the Year    2006   AL       Jim  Leyland
4  johnsda02  TSN Manager of the Year    1997   AL     Davey  Johnson
5  johnsda02  TSN Manager of the Year    2012   NL     Davey  Johnson

In [104]:
managers.head()

playerid  yearid teamid lgid  inseason   g   w   l  rank plyrmgr
0  wrighha01    1871    BS1   NA         1  31  20  10   3.0       Y
1   woodji01    1871    CH1   NA         1  28  19   9   2.0       Y
2  paborch01    1871    CL1   NA         1  29  10  19   8.0       Y
3  lennobi01    1871    FW1   NA         1  14   5   9   8.0       Y
4  deaneha01    1871    FW1   NA         2   5   2   3   8.0       Y

In [125]:
managers_teams = pd.merge(both_names, managers[['playerid', 'teamid', 'yearid']], on = ['playerid', 'yearid'])

In [126]:
managers_teams = managers_teams[['namefirst', 'namelast', 'yearid', 'awardid', 'teamid']]

In [128]:
managers_teams = pd.merge(managers_teams, teams[['teamid', 'name', 'yearid']], on = ['teamid', 'yearid'])

In [137]:
managers_teams['full_name'] = managers_teams['namefirst'] + ' ' + managers_teams['namelast']

In [139]:
managers_teams = managers_teams[['full_name', 'yearid', 'awardid', 'name']]

In [141]:
managers_teams.columns = ['name', 'year', 'award', 'team_name']

In [142]:
managers_teams

name  year                    award             team_name
0    Jim Leyland  1988  TSN Manager of the Year    Pittsburgh Pirates
1    Jim Leyland  1990  TSN Manager of the Year    Pittsburgh Pirates
2    Jim Leyland  1992  TSN Manager of the Year    Pittsburgh Pirates
3    Jim Leyland  2006  TSN Manager of the Year        Detroit Tigers
4  Davey Johnson  1997  TSN Manager of the Year     Baltimore Orioles
5  Davey Johnson  2012  TSN Manager of the Year  Washington Nationals

### 5. Analyze all the colleges in the state of Tennessee. a. Which college has had the most success in the major leagues? b. Use whatever metric for success you like - number of players, number of games, salaries, world series wins, etc.

In [147]:
schools_tn = schools[schools['schoolstate'] == 'TN']

In [148]:
tn_players = pd.merge(schools_tn, collegeplaying, on='schoolid')

In [151]:
tn_players = tn_players.drop_duplicates('playerid')

In [153]:
tn_players.head()

schoolid                    schoolname   schoolcity schoolstate  \
0   austinpeay  Austin Peay State University  Clarksville          TN   
4   austinpeay  Austin Peay State University  Clarksville          TN   
9   austinpeay  Austin Peay State University  Clarksville          TN   
11  austinpeay  Austin Peay State University  Clarksville          TN   
13  austinpeay  Austin Peay State University  Clarksville          TN   

   schoolnick   playerid  yearid  
0         USA  ellisaj01    2000  
4         USA  kellesh01    2003  
9         USA  reynoma02    2006  
11        USA  sherrge01    1998  
13        USA  stewaji01    1959

In [155]:
appearances = pd.read_sql('SELECT * FROM appearances;', con=engine)

In [156]:
appearances.head()

yearid teamid lgid   playerid  g_all  gs  g_batting  g_defense  g_p  g_c  \
0    1871    TRO   NA  abercda01      1 NaN          1          1    0    0   
1    1871    RC1   NA   addybo01     25 NaN         25         25    0    0   
2    1871    CL1   NA  allisar01     29 NaN         29         29    0    0   
3    1871    WS3   NA  allisdo01     27 NaN         27         27    0   27   
4    1871    RC1   NA  ansonca01     25 NaN         25         25    0    5   

   ...  g_2b  g_3b  g_ss  g_lf  g_cf  g_rf  g_of  g_dh  g_ph  g_pr  
0  ...     0     0     1     0     0     0     0   NaN   NaN   NaN  
1  ...    22     0     3     0     0     0     0   NaN   NaN   NaN  
2  ...     2     0     0     0    29     0    29   NaN   NaN   NaN  
3  ...     0     0     0     0     0     0     0   NaN   NaN   NaN  
4  ...     2    20     0     1     0     0     1   NaN   NaN   NaN  

[5 rows x 21 columns]

In [157]:
tn_majors = pd.merge(tn_players[['playerid', 'schoolname']], appearances[['playerid', 'g_all']] )

In [175]:
tn_majors_player_total = tn_majors.groupby(['playerid', 'schoolname']).sum().reset_index()

In [176]:
tn_majors_player_total

playerid                  schoolname  g_all
0    alexada01            Milligan College    662
1    allenfr01              Rhodes College    181
2    altmage01  Tennessee State University    991
3    alvarpe01       Vanderbilt University    851
4    anderda02       University of Memphis    873
..         ...                         ...    ...
157  willido01     University of Tennessee     11
158  williea01           Maryville College      3
159  willimi01       Vanderbilt University    144
160  wrighcl01       Carson-Newman College    360
161   zeidjo01       Vanderbilt University     48

[162 rows x 3 columns]

In [178]:
tn_game_total = tn_majors_player_total.groupby('schoolname').sum().reset_index()

In [181]:
tn_game_total.sort_values('g_all', ascending=False)

schoolname  g_all
28                        University of Tennessee  16841
31                          Vanderbilt University   6215
27                          University of Memphis   3250
0                    Austin Peay State University   2931
23                     Tennessee State University   2284
17                              Maryville College   1589
18              Middle Tennessee State University   1340
30                        University of the South   1194
5                Columbia State Community College   1048
7                 East Tennessee State University    974
20                 Motlow State Community College    967
2                           Carson-Newman College    945
21                                 Rhodes College    719
19                               Milligan College    664
26                               Union University    651
12                             Lambuth University    593
15                    Lincoln Memorial University    534
16                            Lipscomb University    442
25                     Tennessee Wesleyan College    376
13                                   Lane College    365
10                Jackson State Community College    346
11                                   King College    318
29  University of Tennessee Health Science Center    248
22          Southwest Tennessee Community College    222
8                       Freed-Hardeman University    188
1                              Belmont University    184
4               Cleveland State Community College    166
24             Tennessee Technological University     84
14                           LeMoyne-Owen College     72
6                           Cumberland University     62
3   Chattanooga State Technical Community College     61
9                                Hiwassee College     34

### 6. a. Is there any correlation between number of wins and team salary? b. Use data from 2000 and later to answer this question. c. As you do this analysis, keep in mind that salaries across the whole league tend to increase together, so you may want to look on a year-by-year basis.

In [234]:
salaries = pd.read_sql('SELECT * FROM salaries;', con=engine)

In [235]:
salaries = salaries[salaries['yearid'] >= 2000]

In [236]:
team_salaries = salaries.groupby(['teamid', 'yearid']).sum()

In [237]:
team_salaries

salary
teamid yearid             
ANA    2000     51464167.0
       2001     47535167.0
       2002     61721667.0
       2003     79031667.0
       2004    100534667.0
...                    ...
WAS    2012     80855143.0
       2013    113703270.0
       2014    131983680.0
       2015    155587472.0
       2016    141652646.0

[510 rows x 1 columns]

In [238]:
teams_2000 = teams[teams['yearid'] >= 2000]

In [239]:
teams_2000 = teams_2000[['teamid', 'yearid', 'w']]

In [240]:
teams_2000 = teams_2000.groupby(['teamid', 'yearid']).sum()

In [241]:
teams_2000

w
teamid yearid    
ANA    2000    82
       2001    75
       2002    99
       2003    77
       2004    92
...            ..
WAS    2012    98
       2013    86
       2014    96
       2015    83
       2016    95

[510 rows x 1 columns]

In [249]:
teams_2000_salaries = pd.merge(teams_2000, team_salaries, on = ['teamid', 'yearid'], how = 'left')

In [263]:
teams_2000_salaries.reset_index()

teamid  yearid   w       salary
0      ANA    2000  82   51464167.0
1      ANA    2001  75   47535167.0
2      ANA    2002  99   61721667.0
3      ANA    2003  77   79031667.0
4      ANA    2004  92  100534667.0
..     ...     ...  ..          ...
505    WAS    2012  98   80855143.0
506    WAS    2013  86  113703270.0
507    WAS    2014  96  131983680.0
508    WAS    2015  83  155587472.0
509    WAS    2016  95  141652646.0

[510 rows x 4 columns]

In [267]:
teams_2000_salaries.groupby('yearid')[['salary', 'w']].corr()

salary         w
yearid                           
2000   salary  1.000000  0.331090
       w       0.331090  1.000000
2001   salary  1.000000  0.320648
       w       0.320648  1.000000
2002   salary  1.000000  0.442416
       w       0.442416  1.000000
2003   salary  1.000000  0.418704
       w       0.418704  1.000000
2004   salary  1.000000  0.541015
       w       0.541015  1.000000
2005   salary  1.000000  0.491070
       w       0.491070  1.000000
2006   salary  1.000000  0.536276
       w       0.536276  1.000000
2007   salary  1.000000  0.491055
       w       0.491055  1.000000
2008   salary  1.000000  0.326973
       w       0.326973  1.000000
2009   salary  1.000000  0.476310
       w       0.476310  1.000000
2010   salary  1.000000  0.368869
       w       0.368869  1.000000
2011   salary  1.000000  0.409203
       w       0.409203  1.000000
2012   salary  1.000000  0.190057
       w       0.190057  1.000000
2013   salary  1.000000  0.299579
       w       0.299579  1.000000
2014   salary  1.000000  0.322433
       w       0.322433  1.000000
2015   salary  1.000000  0.238783
       w       0.238783  1.000000
2016   salary  1.000000  0.636288
       w       0.636288  1.000000

### 7. It is thought that since left-handed pitchers are more rare, causing batters to face them less often, that they are more effective. Investigate this claim and present evidence to either support or dispute this claim. a. First, determine just how rare left-handed pitchers are compared with right-handed pitchers. 

In [291]:
pitching = pd.read_sql('SELECT * FROM pitching;', con=engine)

In [348]:
halloffame = pd.read_sql('SELECT * FROM halloffame;', con=engine)

In [293]:
awardsplayers = pd.read_sql('SELECT * FROM awardsplayers;', con=engine)

In [298]:
people.shape

(19112, 24)

In [305]:
pitching.shape

(44963, 30)

In [306]:
pitchers.shape

(9302,)

In [307]:
halloffame.shape

(4156, 9)

In [315]:
pitchers = pitching.drop_duplicates('playerid')
pitchers.shape

(9302, 30)

In [316]:
pitchers = pd.merge(people[['playerid', 'throws']], pitchers[['playerid']], on='playerid')

In [317]:
pitchers

playerid throws
0     aardsda01      R
1      aasedo01      R
2      abadfe01      L
3     abbeybe01      R
4     abbeych01      L
...         ...    ...
9297  zoldasa01      L
9298  zuberbi01      R
9299  zumayjo01      R
9300  zuverge01      R
9301   zychto01      R

[9302 rows x 2 columns]

In [320]:
throws = pitchers.groupby('throws')[['playerid']].count()

In [321]:
throws = throws.reset_index()
throws

throws  playerid
0      L      2477
1      R      6605
2      S         1

In [322]:
left = throws[throws['throws'] == 'L']['playerid'].values[0]
right = throws[throws['throws'] == 'R']['playerid'].values[0]

In [324]:
percent_left = (left/(left+right))*100

In [325]:
percent_left

27.273728253688617

### b. Are left-handed pitchers more likely to win the Cy Young Award?

In [335]:
pitchers_award = pd.merge(pitchers, awardsplayers[awardsplayers['awardid'] == 'Cy Young Award'][['playerid', 'awardid']])

In [336]:
pitchers_award

playerid throws         awardid
0    arrieja01      R  Cy Young Award
1    bedrost01      R  Cy Young Award
2     bluevi01      L  Cy Young Award
3    carltst01      L  Cy Young Award
4    carltst01      L  Cy Young Award
..         ...    ...             ...
107  vuckope01      R  Cy Young Award
108   webbbr01      R  Cy Young Award
109  welchbo01      R  Cy Young Award
110   wynnea01      R  Cy Young Award
111   zitoba01      L  Cy Young Award

[112 rows x 3 columns]

In [337]:
pitchers_award_throws = pitchers_award.groupby('throws')[['playerid']].count()

In [343]:
pitchers_award_throws = pitchers_award_throws.reset_index()
pitchers_award_throws

throws  playerid
0      L        37
1      R        75

In [344]:
award_left = pitchers_award_throws[pitchers_award_throws['throws'] == 'L']['playerid'].values[0]
award_right = pitchers_award_throws[pitchers_award_throws['throws'] == 'R']['playerid'].values[0]
award_percent_left = (award_left/(award_left+award_right))*100
award_percent_left

33.035714285714285

### c. Are they more likely to make it into the hall of fame?

In [349]:
pitchers_hall = pd.merge(pitchers, halloffame[halloffame['inducted'] == 'Y'][['playerid', 'inducted']])

In [350]:
pitchers_hall

playerid throws inducted
0    alexape01      R        Y
1    ansonca01      R        Y
2    becklja01      L        Y
3    bendech01      R        Y
4    blylebe01      R        Y
..         ...    ...      ...
96   willivi01      R        Y
97   wrighge01      R        Y
98   wrighha01      R        Y
99    wynnea01      R        Y
100  youngcy01      R        Y

[101 rows x 3 columns]

In [351]:
pitchers_hall_throws = pitchers_hall.groupby('throws')[['playerid']].count()

In [353]:
pitchers_hall_throws = pitchers_hall_throws.reset_index()
pitchers_hall_throws

throws  playerid
0      L        23
1      R        78

In [354]:
hall_left = pitchers_hall_throws[pitchers_hall_throws['throws'] == 'L']['playerid'].values[0]
hall_right = pitchers_hall_throws[pitchers_hall_throws['throws'] == 'R']['playerid'].values[0]
hall_percent_left = (hall_left/(hall_left+hall_right))*100
hall_percent_left

22.772277227722775